### PR TITLE
Mask ACR token on Windows when System.Debug is true

### PR DESCRIPTION
**Pull request by DmitriiBobreshev**
**Description**:
-- Added refresh token to secret masker
-- Added Knob to use --password-stdin on windows

**Related WI**: [AB#2174985](https://dev.azure.com/mseng/b924d696-3eae-4116-8443-9a18392d8544/_workitems/edit/2174985)

**Testing**: 

Tested with ACR token on test tenant. Refresh-token masked if we use --password argument:
image
And passed via stdin if the knob is enabled
image

### CODE REVIEW

File: src/Agent.Sdk/Knob/AgentKnobs.cs [hunk 1]
```diff
@@ -716,5 +716,14 @@ public class AgentKnobs
             "Checks if the PSModulePath environment variable contains locations specific to PowerShell Core.",
             new EnvironmentKnobSource("AZP_AGENT_CHECK_PSMODULES_LOCATIONS"),
             new BuiltInDefaultKnobSource("false"));
+
+        public static readonly Knob UseDockerStdinPasswordOnWindows = new Knob(
+            nameof(CheckPsModulesLocations),
```
Comment (aleksandrlevochkin) [inline src/Agent.Sdk/Knob/AgentKnobs.cs:721]: You forgot to change it probably

**Review by aleksandrlevochkin — CHANGES_REQUESTED** [review]

File: src/Agent.Sdk/Knob/AgentKnobs.cs [hunk 2]
```diff
@@ -716,5 +716,14 @@ public class AgentKnobs
...
+            "If true, use --password-stdin for docker login on Windows.",
+            new EnvironmentKnobSource("AZP_AGENT_USE_DOCKER_STDIN_PASSWORD_WINDOWS"),
+            new RuntimeKnobSource("AZP_AGENT_USE_DOCKER_STDIN_PASSWORD_WINDOWS"),
+            new EnvironmentKnobSource("AZP_AGENT_USE_DOCKER_STDIN_PASSWORD_WINDOWS"),
+            new PipelineFeatureSource("UseDockerStdinPasswordOnWindows"),
+            new BuiltInDefaultKnobSource("false"));
```
Comment (KonstantinTyukalov) [inline src/Agent.Sdk/Knob/AgentKnobs.cs:727]:
```suggestion
            new EnvironmentKnobSource("AZP_AGENT_USE_DOCKER_STDIN_PASSWORD_WINDOWS"),
            new RuntimeKnobSource("AZP_AGENT_USE_DOCKER_STDIN_PASSWORD_WINDOWS"),
            new PipelineFeatureSource("UseDockerStdinPasswordOnWindows"),
            new BuiltInDefaultKnobSource("false"));
```

P.S. If this is FF-only change, are we actually need to have system env variable source?

Comment (KonstantinTyukalov) [reply src/Agent.Sdk/Knob/AgentKnobs.cs:727]: Or, if this something which user should decide by himself, then I believe we don't need the `PipelineFeatureSource`

File: src/Agent.Sdk/Knob/AgentKnobs.cs (hunk 1 above)
Comment (DmitriiBobreshev) [reply src/Agent.Sdk/Knob/AgentKnobs.cs]: fixed, thanks!

File: src/Agent.Sdk/Knob/AgentKnobs.cs (hunk 2 above)
Comment (DmitriiBobreshev) [reply src/Agent.Sdk/Knob/AgentKnobs.cs]: Removed EnvironmentKnobSource, thanks

**Review by KonstantinTyukalov — APPROVED** [review]

**Review by aleksandrlevochkin — APPROVED** [review]

**Review by vmapetr — APPROVED** [review]

**Review by DergachevE — APPROVED** [review]

**Pull request merged**